### ISSUE TITLE
Fix S2589 FP/FN: Unsigned number doubling by adding with itself inside while loop

### ISSUE THREAD

**Issue by MineCake147E** [SonarSource/sonar-dotnet#9184]
### Description

[S2589](https://sonarsource.github.io/rspec/#/rspec/S2589/csharp) is being reported when using unsigned `x += x`.

### Repro steps

```csharp
public static void UIntPtrS2589FalsePositive()
{
    nuint u = 1;
    while (u > 0)   // noncompliant FP
    {
        u += u;     // this addition could overflow to 0
    }
}

public static void UInt64S2589FalsePositive()
{
    ulong u = 1;
    while (u > 0)   // noncompliant FP
    {
        u += u;     // this addition could overflow to 0
    }
}

public static void UInt32S2589FalsePositive()
{
    uint u = 1;
    while (u > 0)   // noncompliant FP
    {
        u += u;     // this addition could overflow to 0
    }
}
```

### Expected behavior

No issue should be found.

### Actual behavior

`u > 0` gets [S2589](https://sonarsource.github.io/rspec/#/rspec/S2589/csharp)

### Known workarounds

Use either `u *= 2;` or `u <<= 1;` instead of `u += u;`.

### Related information

* SonarLint for Visual Studio 2022 7.8.0.88494
* Microsoft Visual Studio Community 2022 (64-bit) - Preview Version 17.10.0 Preview 4.0
* dotnet 8.0.300-preview.24203.14
* Operating System: Windows 11 Home 22H2 22621.3447

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1591](https://sonarsource.atlassian.net/browse/NET-1591)

[NET-1591]: https://sonarsource.atlassian.net/browse/NET-1591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
